src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalProductOpt.java

```diff
@@ -33,22 +34,20 @@ public LocalProductOpt(BaseDTO dto) {
     }
 
     public void getSupportedLocales(MessageCacheItem cacheItem) {
-        List<String> supportedLocales = new ArrayList<String>();
+        List<String> supportedLocales = new ArrayList<>();
         try {
-
             Path path = Paths.get(VIPCfg.getInstance().getOfflineResourcesBaseUrl());
-
-            URI uri = Thread.currentThread().getContextClassLoader().
-                    getResource(path.toString()).toURI();
-
-            if (uri.getScheme().equals("jar")) {
-                try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
-                    path = fileSystem.getPath(path.toString());
+            List<URI> uris = FileUtil.findUris(path);
+            for (URI uri : uris) {
+                if (uri.getScheme().equals("jar")) {
+                    try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
+                        path = fileSystem.getPath(Paths.get(uri).toString());
+                        getSupportedLocales(path, supportedLocales);
+                    }
+                } else {
+                    path = Paths.get(uri);
                     getSupportedLocales(path, supportedLocales);
                 }
```

Comment:
Here should only get the supported locales of main application, right? Because supported locales should oriented on main application.

If yes, when the bundle's root path of shared lib and that of main application's is the same and their supported locales are different, these code will merge shared lib's supported locale and main application's, which lead to when 'getAllComponentTranslation' method in ProductService is executed, there will be some wrong locale&component combinations.

Reply:
Yes, you are right. For now, getSupportedLocales will include all locales under all components under the offlineResourceBaseURL (whether inside main app or dependency jars). However, a wrong locale&component combination may happen even without a shared lib. For instance, 2 components in the main app may have different sets of bundles at some point (e.g. component A's translations are ready but component B's translations are not ready yet), so they have different supported locales. This also give a false positive in some combinations like what you said, and so will still try to fetch for the bundle even if the locale does not exist under the component.

We have been ignoring this issue even before this PR. We should improve this separately. In my opinion, getSupportedLocales should be fetched and cached per component, which will resolve the issue. This should apply to components in both the main app and shared libs. This has to be done outside of this PR.

Reply:
I think 2 components in the main app have different sets of bundles can only happen at development stage, rarely happen at production stage, so wrong locale&component combination rarely happen at production stage, too. So before this PR we can ignore the issue. 

But after this PR, wrong locale&component combination may happen at production stage. So there may be necessary to improve this.

I think there maybe no necessary to getSupportedLocales per product component, add {product}/{version} floder in local bundle path is enough, how do you think?

Reply:
No we cannot ignore the issue. It could happen that there are multiple shared libraries (hence, many URIs) using the same product-version, but different component-locales set in each shared library. This happens in production. If we do not get all of those URIs, we do not get the complete.

Again, let's resolve the issue of getting supported locales and components by product-version in #870

Reply:
> I agree, the correct solution should be getting supported locales by product and version, not get them from all product-versions.

Yes, I agree. We will work on that separately.